src/main/java/edu/texas/social_computing/hospitals/RHRP.java

```diff
@@ -0,0 +1,134 @@
+        return run(hospitalTable, residentTable, freeResidents, new Matching());
+    }
+
+    public static void main(String[] args) throws FileNotFoundException {
+        List<Resident> residents = FileImporter.importResidents(args[0]);
+        List<Hospital> hospitals = FileImporter.importHospitals(args[1]);
+        System.out.println("loaded Files");
+        System.out.println("residents: " + Integer.toString(residents.size()));
+        System.out.println("hospitals: " + Integer.toString(hospitals.size()));
+
+        // make lookup tables
+        HospitalTable hospitalTable = HospitalTable.create(hospitals);
+        ResidentTable residentTable = ResidentTable.create(residents);
+        System.out.println("generated tables");
+
+        // call HRP
+        Matching m = RHRP.run(hospitalTable, residentTable, new ArrayDeque<>(residentTable.getAll()));
+        System.out.println(m);
+
+
+        Set<String> unassignedIds = m.getAllUnassigned(residentTable.getAll()).stream()
+                .map(Resident::getId)
+                .collect(ImmutableSet.toImmutableSet());
+        System.out.println("Unassigned: " + unassignedIds);
+    }
+
+    private static List<String> getNonRankedIds(Hospital h, List<Resident> residents) {
+        return residents.stream()
+                .filter(res -> h.rankOf(res) == Integer.MAX_VALUE)
+                .map(Resident::getId)
+                .collect(ImmutableList.toImmutableList());
+    }
+
+    private static void tryToMatch(Hospital h, Resident resident,
+                                   ResidentTable residentTable, Queue<Resident> freeResidents, Matching m) {
+        // Only if the hospital has this resident on its pref list can it accept the resident
+        if (h.getPreferences().contains(resident.getId())) {
+            m.assign(resident, h);
+            if (m.isOverSubscribed(h)) {
+                Resident worstResident = m.getWorstAssignedResident(h);
+                m.unassign(worstResident);
+                if (!worstResident.getPreferences().isEmpty() && !worstResident.equals(resident)) {
+                    freeResidents.add(worstResident);
+                }
+                if (worstResident.hasPartner()) {
+                    Resident worstResidentPartner = residentTable.getResidentById(worstResident.getPartnerId());
+                    m.unassign(worstResidentPartner);
+                    if (!worstResidentPartner.getPreferences().isEmpty()
+                            && !worstResident.equals(resident)) {
+                        // FIXME Problem here! During the partner's call worstResident.equals(resident) check
+                        //  needs to be modified
+                        freeResidents.add(worstResidentPartner);
+                    }
+                }
+            }
+        } else {
+            if (resident.hasPartner()) {
+                if (m.hasAssignment(residentTable.getResidentById(resident.getPartnerId()))) {
```

Comment:
`residentTable.getResidentById(resident.getPartnerId())` gets called 3 times. you should probably use
`Resident partner = residentTable.getResidentById(resident.getPartnerId());`